sdk/lib/opentelemetry/sdk/resources/auto_detector.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/sdk/resources/detectors/google_cloud_platform'
+require 'opentelemetry/sdk/resources/detectors/telemetry'
+
+module OpenTelemetry
+  module SDK
+    module Resources
+      class AutoDetector
+        DETECTORS = [
+          OpenTelemetry::SDK::Resources::Detectors::GoogleCloudPlatform.new,
+          OpenTelemetry::SDK::Resources::Detectors::Telemetry.new,
+        ]
+
+        def initialize(initial_resource = OpenTelemetry::SDK::Resources::Resource.create)
+          @initial_resource = initial_resource
+        end
+
+        def detect
+          DETECTORS.each do |detector|
+            new_resource = detector.detect();
+            @initial_resource = @initial_resource.merge(new_resource)
```

Comment:
Is there a good reason to carry this state around and `.merge` inside this method rather than just returning the auto-detected `Resource` and merging outside?

Reply:
No good reason, have changed it to be a bit simpler.